src/thunderbots/software/ai/world/world.cpp

```diff
@@ -32,25 +35,25 @@ World::World(const Field &field, const Ball &ball, const Team &friendly_team,
 
 void World::updateFieldGeometry(const Field &new_field_data)
 {
-    field_.updateDimensions(new_field_data);
+    field_ = new_field_data;
     updateTimestamp(getMostRecentTimestampFromMembers());
 }
 
 void World::updateBallState(const Ball &new_ball_data)
 {
-    ball_.updateState(new_ball_data);
+    ball_ = new_ball_data;
     updateTimestamp(getMostRecentTimestampFromMembers());
 }
 
 void World::updateFriendlyTeamState(const Team &new_friendly_team_data)
 {
-    friendly_team_.updateState(new_friendly_team_data);
+    friendly_team_ = new_friendly_team_data;
     updateTimestamp(getMostRecentTimestampFromMembers());
 }
 
 void World::updateEnemyTeamState(const Team &new_enemy_team_data)
 {
-    enemy_team_.updateState(new_enemy_team_data);
+    enemy_team_ = new_enemy_team_data;
```

Comment:
What was wrong with what we were doing before? Should we remove the `updateState` methods?

Reply:
It was a mix of making most update functions function like assignment (which was being done in `ai/main.cpp` before) but let some update state. It was a change necessary to act like we're "copying" the world data into the `ai_logic` node, and only update the `gamestate` state.

I think we should revisit the usage of the update functions when we create the sensor fusion abstraction. I'd prefer to leave these as they are for now unless we have a good reason to change them immediately.

Reply:
I'm still confused what the issue is with calling `updateState` on our member variables here?

Reply:
After some discussion offline and some thought, I don't think we should make this change. I have what is essentially a fix for this issue in an upcoming PR, and this obfuscates the concept of "updating" something.